integrations/azure-devops/azure_devops/client/azure_devops_client.py

```diff
@@ -28,13 +28,35 @@ def create_from_ocean_config(cls) -> "AzureDevopsClient":
         event.attributes["azure_devops_client"] = azure_devops_client
         return azure_devops_client
 
+    async def get_single_project(self, project_id: str) -> dict[str, Any]:
+        project_url = (
+            f"{self._organization_base_url}/{API_URL_PREFIX}/projects/{project_id}"
+        )
+        project = (await self.send_request("GET", project_url)).json()
+        return project
+
     @cache_iterator_result()
-    async def generate_projects(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+    async def generate_projects(
+        self, sync_default_team: bool = False
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        """
+        sync_default_team:bool - The List projects endpoint of ADO API excludes default team of a project.
+        By setting leveraging the sync_default_team flag, we optionally fetch the default team from the get project
+        endpoint using the project id which we obtain from the list projects endpoint.
+        read more -> https://learn.microsoft.com/en-us/rest/api/azure/devops/core/projects/list?view=azure-devops-rest-7.1&tabs=HTTP#teamprojectreference
+        """
+
         params = {"includeCapabilities": "true"}
         projects_url = f"{self._organization_base_url}/{API_URL_PREFIX}/projects"
         async for projects in self._get_paginated_by_top_and_continuation_token(
             projects_url, additional_params=params
         ):
+            if sync_default_team:
+                logger.info("Adding default team to projects")
+                projects = [
+                    await self.get_single_project(project["id"]) for project in projects
+                ]
```

Comment:
this can be done using asyncio gather or semaphore, if it returns big batches of projects, so it won't take too long